### PR TITLE
Use iree_string_view_atoi_int32 in CUDA driver module flag parsing.

### PR DESCRIPTION
**Pull request by okkwon**
use iree_string_view_atoi_int32() to parse strings into int32_t.

### CODE REVIEW

**Review by benvanik — CHANGES_REQUESTED** [review]
We have this already under iree_string_view_atoi_int32 - you can use that instead

File: runtime/src/iree/hal/drivers/cuda/registration/driver_module.c [hunk 1]
```diff
@@ -53,12 +53,14 @@ static iree_status_t iree_hal_cuda_init_nccl_rank_and_count(
   if (!nprocs_str) {
     return iree_ok_status();
   }
-
-  int nprocs = atoi(nprocs_str);
-  if (nprocs <= 0) {
+  iree_string_view_t nprocs_sv = {nprocs_str, strlen(nprocs_str)};
+  bool parsed;
+  int32_t nprocs = -1;
+  parsed = iree_string_view_atoi_int32(nprocs_sv, &nprocs);
```
Comment (benvanik) [inline runtime/src/iree/hal/drivers/cuda/registration/driver_module.c]:
```suggestion
  int32_t nprocs = -1;
  bool parsed = iree_string_view_atoi_int32(iree_make_cstring_view(nprocs_str), &nprocs);
```

(always default initialize values)

File: runtime/src/iree/hal/drivers/cuda/registration/driver_module.c [hunk 2]
```diff
@@ -70,8 +72,10 @@ static iree_status_t iree_hal_cuda_init_nccl_rank_and_count(
         IREE_STATUS_INVALID_ARGUMENT,
         "IREE_CUDA_NCCL_PROCID must be set when IREE_CUDA_NCCL_NPROCS is set.");
   }
-  int procid = atoi(procid_str);
-  if (procid < 0 || procid >= nprocs) {
+  iree_string_view_t procid_sv = {procid_str, strlen(procid_str)};
+  int32_t procid = -1;
+  parsed = iree_string_view_atoi_int32(procid_sv, &procid);
+  if (!parsed || procid < 0 || procid >= nprocs) {
```
Comment (benvanik) [inline runtime/src/iree/hal/drivers/cuda/registration/driver_module.c]:
```suggestion
  int32_t procid = -1;
  parsed = iree_string_view_atoi_int32(iree_make_cstring_view(procid_str), &procid);
  if (!parsed || procid < 0 || procid >= nprocs) {
```

**Review by benvanik — CHANGES_REQUESTED** [review]

**Review by benvanik — COMMENTED** [review]
(I also changed the issue title - please try to use titles that are descriptive of what the change does)

**Comment by okkwon** [comment]
Thanks for the suggestion @benvanik!

**Review by benvanik — APPROVED** [review]

**Pull request merged**